interfaces/repo.go

```diff
@@ -400,70 +408,28 @@ func (r *Repository) disconnect(plug *Plug, slot *Slot) {
 	if len(r.plugSlots[plug]) == 0 {
 		delete(r.plugSlots, plug)
 	}
-}
-
-// ConnectedSlots returns all the plugs connected to a given snap.
-func (r *Repository) ConnectedSlots(snapName string) map[*Slot][]*Plug {
-	r.m.Lock()
-	defer r.m.Unlock()
-
-	result := make(map[*Slot][]*Plug)
-	for _, slot := range r.slots[snapName] {
-		for plug := range r.slotPlugs[slot] {
-			result[slot] = append(result[slot], plug)
+	for i := 0; i < len(slot.Connections); i++ {
+		plugRef := slot.Connections[i]
+		if plugRef.Snap == plug.Snap && plugRef.Name == plug.Name {
```

Comment:
Can we please have an RefersTo method on plugRef:

    if plugRef.RefersTo(plug) {